File: PieGallery/app/src/main/java/com/deadely/piegallery/base/BaseFragment.java
package com.deadely.piegallery.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.deadely.piegallery.App;
import com.deadely.piegallery.di.component.FragmentComponent;
import com.deadely.piegallery.di.module.FragmentModule;

public abstract class BaseFragment extends Fragment {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        FragmentComponent fragmentComponent = App.getInstance().getComponent().fragmentComponent(new FragmentModule(this));
        inject(fragmentComponent);

    }

    protected abstract void inject(FragmentComponent fragmentComponent);
}

File: PieGallery/app/src/main/java/com/deadely/piegallery/base/BaseFragment.kt
package com.deadely.piegallery.base

import android.os.Bundle
import android.view.View
import moxy.MvpAppCompatFragment

abstract class BaseFragment(layout: Int) : MvpAppCompatFragment(layout), BaseView {
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        getExtras()
        setListeners()
        initView()
    }

    abstract fun setListeners()
    abstract fun getExtras()
    abstract fun initView()
}
